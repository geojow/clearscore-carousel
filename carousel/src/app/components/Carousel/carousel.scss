.carousel__container:before {
  content: '';
  position: absolute;
  width: 400px;
  height: 400px;
  filter: blur(10px);
  background-image: url('/background--dashboard.png');
  background-position: 50%;
  border-radius: 50%;
}

.carousel__container {
  display: flex;
  position: relative;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 375px;
  height: 375px;
  border: 1px solid #fff;
  border-radius: 50%;
  overflow: hidden;
}
