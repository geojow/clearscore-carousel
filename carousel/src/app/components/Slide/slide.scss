.slide__container {
  height: 375px;
  width: 375px;
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  font-family: Helvetica, sans-serif;

  color: white
}

p {
  margin: 0;
}

.slide__content-upper, .slide__content-lower {
  font-size: 20px;
  font-weight: 300;
}

.slide__content-middle {
  font-size: 90px;
  font-weight: 100;
}

