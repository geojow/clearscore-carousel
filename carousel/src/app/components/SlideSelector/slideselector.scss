.slide-selector__container {
  list-style: none;
  display: flex;
  position: absolute;
  bottom: 20px;
}


.slide-selector__item:hover {
  background-color: #fff;
  height: 5px;
  width: 5px;
  margin: 0 5px;
  border-radius: 50%;
}

.slide-selector__item-active {
  background-color: #fff;
  border: solid 1px #fff;
  height: 5px;
  width: 5px;
  margin: 0 5px;
  border-radius: 50%;
}



.slide-selector__item {
  border: solid 1px #fff;
  height: 5px;
  width: 5px;
  margin: 0 5px;
  border-radius: 50%;
}


