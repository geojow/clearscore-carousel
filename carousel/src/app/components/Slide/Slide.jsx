import React from 'react'
import bemHelper from '../../utils/bem';
import './slide.scss';

const cn = bemHelper({ block: 'slide' });

const Slide = () => <li className={cn('container')} />

export default Slide;
