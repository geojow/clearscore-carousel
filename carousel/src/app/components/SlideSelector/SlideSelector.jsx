import React from "react";
import PropTypes from "prop-types";
import bemHelper from "../../utils/bem";
import "./slideselector.scss";

const cn = bemHelper({ block: "slide-selector" });

const SlideSelector = ({ numberOfSlides, currentSlide, setSlide }) => (
  <ul className={cn("container")}>
    {[...Array(numberOfSlides).keys()].map((number, index) => (
      <li
        className={`${cn("item")} ${
          currentSlide === index ? `${cn("item-active")}` : ""
        }`}
        key={number}
        onClick={() => setSlide(number)}
      ></li>
    ))}
  </ul>
);

SlideSelector.propTypes = {
  numberOfSlides: PropTypes.number.isRequired,
  currentSlide: PropTypes.number.isRequired,
  setSlide: PropTypes.func.isRequired
};

export default SlideSelector;
