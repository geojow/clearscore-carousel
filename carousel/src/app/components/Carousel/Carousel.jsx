import React, { Component } from 'react';
import PropTypes from 'prop-types'
import bemHelper from '../../utils/bem';
import Slide from '../Slide/Slide'
import './carousel.scss';

const cn = bemHelper({ block: 'carousel' });

class Carousel extends Component {
  state = {
    slideIndex: 0
  }

  getSlides() {
    const { creditReport } = this.props;
    return creditReport ? [{
      upper: "Your credit score is",
      middle: creditReport.score.toString(),
      lower: `out of ${creditReport.maxScoreValue}`
    }, {
      upper: 'Your long term debt total',
      middle: `Â£${creditReport.currentLongTermDebt}`,
      lower: creditReport.changeInLongTermDebt === 0 ? "No change from last month" : creditReport.changeInLongTermDebt < 0 ? `Down ${creditReport.changeInLongTermDebt} from last month` : `Up ${(creditReport.changeInLongTermDebt)} from last month`
    }] : []
  }

  render() {
    const { slideIndex } = this.state;
    const slides = this.getSlides()
    return (
      <ul className={cn('container')}>
        <Slide slide={slides[slideIndex]} />
      </ul>
    )
  }
}

Carousel.propTypes = {
  creditReport: PropTypes.shape({
    score: PropTypes.number.isRequired,
    maxScoreValue: PropTypes.number.isRequired,
    currentLongTermDebt: PropTypes.number.isRequired,
    changeInLongTermDebt: PropTypes.number.isRequired,
  })
}

export default Carousel;
