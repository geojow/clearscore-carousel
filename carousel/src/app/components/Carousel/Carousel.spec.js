import React from "react";
import { shallow, mount } from "enzyme";
import Carousel from "./Carousel";
import Slide from "../Slide/Slide";
import SlideSelector from "../SlideSelector/SlideSelector";

describe("Carousel", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Carousel />)));

  it("should render a <ul />", () => {
    expect(wrapper.find("ul").length).toEqual(1);
  });

  it("should render a Slide and SlideSelector component", () => {
    expect(
      wrapper.containsAllMatchingElements([
        <Slide />,
        <SlideSelector
          numberOfSlides={wrapper.instance().state.numberOfSlides}
          currentSlide={wrapper.instance().state.slideIndex}
          setSlide={wrapper.instance().setIndex}
        />
      ])
    ).toEqual(true);
  });
});

describe("mounted Carousel", () => {
  let wrapper;

  beforeEach(() => (wrapper = mount(<Carousel />)));

  it("it calls setIndex when a SlideSelector is clicked", () => {
    wrapper.setState({
      slides: [{ id: "1", upper: "", middle: "", lower: "", progress: null }],
      numberOfSlides: 1
    });
    const spy = jest.spyOn(wrapper.instance(), "setIndex");
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledTimes(0);
    wrapper
      .find(".slide-selector__item")
      .first()
      .simulate("click");
    expect(spy).toHaveBeenCalledTimes(1);
  });
});

describe("setIndex", () => {
  let wrapper;

  beforeEach(() => (wrapper = mount(<Carousel />)));

  it("it updates slideIndex", () => {
    wrapper.setState({
      slideIndex: 0,
      slides: [
        { id: "1", upper: "", middle: "", lower: "", progress: null },
        { id: "2", upper: "", middle: "", lower: "", progress: null }
      ],
      numberOfSlides: 2
    });
    wrapper.instance().setIndex(1);
    expect(wrapper.state("slideIndex")).toEqual(1);
  });

  it("it does not update slideIndex if you try to set the same index", () => {
    wrapper.setState({
      slideIndex: 1,
      slides: [
        { id: "1", upper: "", middle: "", lower: "", progress: null },
        { id: "2", upper: "", middle: "", lower: "", progress: null }
      ],
      numberOfSlides: 2
    });
    wrapper.instance().setIndex(1);
    expect(wrapper.state("slideIndex")).toEqual(1);
  });
});
