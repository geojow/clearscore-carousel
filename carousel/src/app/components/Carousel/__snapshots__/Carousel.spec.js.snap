// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Carousel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Slide
          slide={undefined}
        />,
        <SlideSelector
          currentSlide={0}
          numberOfSlides={0}
          setSlide={[Function]}
        />,
      ],
      "className": "carousel__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "slide": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentSlide": 0,
          "numberOfSlides": 0,
          "setSlide": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Slide
            slide={undefined}
          />,
          <SlideSelector
            currentSlide={0}
            numberOfSlides={0}
            setSlide={[Function]}
          />,
        ],
        "className": "carousel__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "slide": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentSlide": 0,
            "numberOfSlides": 0,
            "setSlide": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
